src/aggregator/aggregator/elem_base.go

```diff
@@ -731,3 +731,92 @@ func newParsedPipeline(pipeline applied.Pipeline) (parsedPipeline, error) {
 		Rollup:                 rollup,
 	}, nil
 }
+
+// placeholder to make compiler happy about generic elem base
+// NB: lockedAggregationFromPool and not newLockedAggregation to avoid yet another rename hack in makefile
+func lockedAggregationFromPool(
+	aggregation typeSpecificAggregation,
+	sourcesSeen map[uint32]*bitset.BitSet,
+) *lockedAggregation {
+	return &lockedAggregation{
+		aggregation: aggregation,
+		sourcesSeen: sourcesSeen,
+	}
+}
+
+func (l *lockedAggregation) close() {
+	l.aggregation.Close()
+}
+
+var lockedCounterAggregationPool = sync.Pool{New: func() interface{} { return &lockedCounterAggregation{} }}
+
+func lockedCounterAggregationFromPool(
+	aggregation counterAggregation,
+	sourcesSeen map[uint32]*bitset.BitSet,
+) *lockedCounterAggregation {
+	l := lockedCounterAggregationPool.Get().(*lockedCounterAggregation)
+	l.aggregation = aggregation
```

Comment:
Nit: could factor the zeroing into a `reset` method, and reuse here + in close? Not a huge deal though.

Reply:
I simplified and made close() foolproof